### PR TITLE
[CSFix] Diagnose extraneous force unwraps in ambiguous context

### PR DESCRIPTION
**Pull request by xedin**

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please smoke test

**Comment by xedin** [comment]
@swift-ci please smoke test

File: test/StringProcessing/Parse/forward-slash-regex.swift [hunk 1]
```diff
@@ -121,6 +121,7 @@ do {
 } // expected-error {{expected expression after operator}}
 
 _ = /x/??/x/ // expected-error {{'/' is not a postfix unary operator}}
+// expected-error@-1 2 {{annot use optional chaining on non-optional value of type 'Regex<Substring>'}}
```
Comment (xedin) [inline test/StringProcessing/Parse/forward-slash-regex.swift:124]: WDYT, @hamishknight ?

Comment (hamishknight) [reply test/StringProcessing/Parse/forward-slash-regex.swift:124]: I think this makes sense, as it'll be parsed as `(/x/)?? / (x/)`

Comment (xedin) [reply test/StringProcessing/Parse/forward-slash-regex.swift:124]: Thanks!

Comment (xedin) [reply test/StringProcessing/Parse/forward-slash-regex.swift:124]: I need to add `c` I dropped by mistake from `cannot` 🤦

**Comment by xedin** [comment]
@swift-ci please test

**Review by hborla — APPROVED** [review]
Nice!

**Comment by xedin** [comment]
@swift-ci please smoke test macOS platform

**Pull request merged**